rotkehlchen/tests/db/test_db_upgrades.py

```diff
@@ -1161,14 +1161,41 @@ def test_upgrade_db_21_to_22(user_data_dir):  # pylint: disable=unused-argument
     assert db.get_version() == 22
 
 
-def test_upgrade_db_22_to_23(user_data_dir):  # pylint: disable=unused-argument
+def test_upgrade_db_22_to_23_with_frontend_settings(user_data_dir):
     """Test upgrading the DB from version 22 to version 23.
 
-    Migrates the settings entries 'thousand_separator', 'decimal_separator' and
-    'currency_location' into the 'frontend_settings' entry.
+    - Migrates the settings entries 'thousand_separator', 'decimal_separator'
+    and 'currency_location' into the 'frontend_settings' entry.
+    - Deletes Bitfinex trades and their used query range, so trades can be
+    populated again with the right `fee_asset`.
     """
     msg_aggregator = MessagesAggregator()
-    _use_prepared_db(user_data_dir, 'v22_rotkehlchen.db')
+    _use_prepared_db(user_data_dir, 'v22_rotkehlchen_w_frontend_settings.db')
+    db_v22 = _init_db_with_target_version(
+        target_version=22,
+        user_data_dir=user_data_dir,
+        msg_aggregator=msg_aggregator,
+    )
+    cursor = db_v22.conn.cursor()
+
+    # Check all settings exist
```

Comment:
Given the fact the db data before the migration is not visible at a first glance, and we have 2 db files with a subtle change, this is how I like to test it:
- Assert pre-conditions
- Migrate
- Assert changes

Let me know if you like it!

Reply:
Sure it's fine!